### PR TITLE
En 4170 economics parameters

### PR DESCRIPTION
**Pull request by miiu96**
add economics parameters in a .toml file
fix failing unit tests

### CODE REVIEW

File: config/economics.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#Econimics config of the node
```
Comment (SebastianMarian) [inline config/economics.toml]: Why do you need another economics.toml in this package?

File: process/block/preprocess/rewardsHandler.go [hunk 2]
```diff
@@ -73,6 +70,11 @@ func NewRewardTxHandler(
 	if rewardTxPool == nil || rewardTxPool.IsInterfaceNil() {
 		return nil, process.ErrNilRewardTxDataPool
 	}
+	if economicsRewards == nil {
+		return nil, errors.New("nil economics rewards handler")
```
Comment (SebastianMarian) [inline process/block/preprocess/rewardsHandler.go]: Please create a defined error

File: process/economics/economicsData_test.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/process/economics"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEconomicsData_ShouldWork(t *testing.T) {
+	t.Parallel()
+
+	economicsData := economics.NewEconomicsData(&config.ConfigEconomics{})
+
```
Comment (SebastianMarian) [inline process/economics/economicsData_test.go]: Remove empty row

File: process/factory/shard/preProcessorsContainerFactory.go [hunk 4]
```diff
@@ -84,6 +88,9 @@ func NewPreProcessorsContainerFactory(
 	if rewardsProducer == nil || rewardsProducer.IsInterfaceNil() {
 		return nil, process.ErrNilInternalTransactionProducer
 	}
+	if economicsFee == nil {
+		return nil, errors.New("nil economics fee handler")
```
Comment (SebastianMarian) [inline process/factory/shard/preProcessorsContainerFactory.go]: Please create a defined error

File: process/transaction/process.go [hunk 5]
```diff
@@ -72,6 +68,9 @@ func NewTxProcessor(
 	if txTypeHandler == nil || txTypeHandler.IsInterfaceNil() {
 		return nil, process.ErrNilTxTypeHandler
 	}
+	if economicsFee == nil {
+		return nil, errors.New("nil economics fee handler")
```
Comment (SebastianMarian) [inline process/transaction/process.go]: Please create a defined error

File: process/transaction/process.go [hunk 6]
```diff
@@ -72,6 +68,9 @@ func NewTxProcessor(
 	if txTypeHandler == nil || txTypeHandler.IsInterfaceNil() {
 		return nil, process.ErrNilTxTypeHandler
 	}
+	if economicsFee == nil {
```
Comment (SebastianMarian) [inline process/transaction/process.go]: Maybe you can also create IsInterfaceNil method and check in the same way as the other interfaces are checked here for nil

File: process/factory/shard/preProcessorsContainerFactory.go [hunk 7]
```diff
@@ -84,6 +88,9 @@ func NewPreProcessorsContainerFactory(
 	if rewardsProducer == nil || rewardsProducer.IsInterfaceNil() {
 		return nil, process.ErrNilInternalTransactionProducer
 	}
+	if economicsFee == nil {
```
Comment (SebastianMarian) [inline process/factory/shard/preProcessorsContainerFactory.go]: Maybe you can also create IsInterfaceNil method and check in the same way as the other interfaces are checked here for nil

File: process/block/preprocess/rewardsHandler.go [hunk 8]
```diff
@@ -73,6 +70,11 @@ func NewRewardTxHandler(
 	if rewardTxPool == nil || rewardTxPool.IsInterfaceNil() {
 		return nil, process.ErrNilRewardTxDataPool
 	}
+	if economicsRewards == nil {
```
Comment (SebastianMarian) [inline process/block/preprocess/rewardsHandler.go]: Maybe you can also create IsInterfaceNil method and check in the same way as the other interfaces are checked here for nil

File: config/economics.toml (hunk 1 above)
Comment (miiu96) [reply config/economics.toml]: File was removed.

File: process/block/preprocess/rewardsHandler.go (hunk 2 above)
Comment (miiu96) [reply process/block/preprocess/rewardsHandler.go]: Done.

File: process/economics/economicsData_test.go (hunk 3 above)
Comment (miiu96) [reply process/economics/economicsData_test.go]: Done.

File: process/factory/shard/preProcessorsContainerFactory.go (hunk 4 above)
Comment (miiu96) [reply process/factory/shard/preProcessorsContainerFactory.go]: Done.

File: process/transaction/process.go (hunk 5 above)
Comment (miiu96) [reply process/transaction/process.go]: Done.

File: process/block/preprocess/rewardsHandler.go (hunk 8 above)
Comment (miiu96) [reply process/block/preprocess/rewardsHandler.go]: Done.

File: process/factory/shard/preProcessorsContainerFactory.go (hunk 7 above)
Comment (miiu96) [reply process/factory/shard/preProcessorsContainerFactory.go]: Done.

File: process/transaction/process.go (hunk 6 above)
Comment (miiu96) [reply process/transaction/process.go]: Done.

File: process/errors.go [hunk 9]
```diff
@@ -490,5 +475,11 @@ var ErrOverallBalanceChangeFromSC = errors.New("SC output balance updates are wr
 // ErrNilTxsPoolsCleaner signals that a nil transactions pools cleaner has been provided
 var ErrNilTxsPoolsCleaner = errors.New("nil transactions pools cleaner")
 
-// ErrZeroMaxCleanTime signal that cleaning time for pools its less or equals with 0
+// ErrZeroMaxCleanTime signals that cleaning time for pools its less or equals with 0
```
Comment (AdoAdoAdo) [inline process/errors.go]: **is** less or equal

File: cmd/node/config/economics.toml [hunk 10]
```diff
@@ -0,0 +1,18 @@
+#Econimics config of the node
```
Comment (AdoAdoAdo) [inline cmd/node/config/economics.toml]: typo Economics

**Pull request merged**